src/nisarqa/utils/metadata_checks.py

```diff
@@ -168,102 +166,204 @@ def verify_metadata_cubes(
+        # there are corresponding datasets with x coordinates and y coordinates
+        # of the correct length. If these elements are missing, exceptions
+        # will get thrown.
         spec = nisarqa.Version.from_string(product.product_spec_version)
 
-        # Non-InSAR products have calibrationInformation groups
-        if isinstance(product, nisarqa.NonInsarProduct):
-            for freq in product.freqs:
-
-                if spec >= nisarqa.Version(1, 1, 0):
-                    for cube in product.neb_metadata_cubes(freq):
-                        has_finite &= _metadata_cube_has_finite_pixels(cube)
-                        passes &= has_finite
-                        passes &= _metadata_cube_has_finite_pixels(cube)
-                        passes &= _metadata_cube_is_not_all_zeros(cube)
-                        passes &= _check_gdal(c=cube)
-
-                    for cube in product.elevation_antenna_pat_metadata_cubes(
-                        freq
-                    ):
-                        has_finite &= _metadata_cube_has_finite_pixels(cube)
-                        passes &= has_finite
-                        passes &= _metadata_cube_is_not_all_zeros(cube)
-                        passes &= _check_gdal(c=cube)
-
-                    if isinstance(product, nisarqa.RSLC):
-                        for cube in product.geometry_metadata_cubes():
-                            has_finite &= _metadata_cube_has_finite_pixels(cube)
-                            passes &= has_finite
-                            passes &= _metadata_cube_is_not_all_zeros(cube)
-
-                        for cube in product.crosstalk_metadata_cubes():
-                            has_finite &= _metadata_cube_has_finite_pixels(cube)
-                            passes &= has_finite
-                            passes &= _metadata_cube_is_not_all_zeros(cube)
+        for freq in product.freqs:
+
+            if spec < nisarqa.Version(1, 1, 0):
+                break
+
+            for ds in product.metadata_neb_datasets(freq):
+                has_finite &= _dataset_has_finite_pixels(ds)
+                passes &= has_finite
+                passes &= _dataset_has_finite_pixels(ds)
+                passes &= _dataset_is_not_all_zeros(ds)
+                passes &= _check_gdal(product=product, ds=ds)
+
+            for ds in product.metadata_elevation_antenna_pat_datasets(freq):
+                has_finite &= _dataset_has_finite_pixels(ds)
+                passes &= has_finite
+                passes &= _dataset_is_not_all_zeros(ds)
+                passes &= _check_gdal(product=product, ds=ds)
+
+            if isinstance(product, nisarqa.SLC):
+                for ds in product.metadata_geometry_datasets():
+                    has_finite &= _dataset_has_finite_pixels(ds)
+                    passes &= has_finite
+                    passes &= _dataset_is_not_all_zeros(ds)
+                    passes &= _check_gdal(product=product, ds=ds)
```

Comment:
Let's avoid some code duplication by chaining all of these iterators together and looping over the whole thing just once.

(Sorry for the indentation -- please feel free to run black to clean up the formatting.)

```python
from itertools import chain
```

```suggestion
            calib_datasets = chain(
                [product.metadata_neb_datasets(freq), 
                 product.metadata_elevation_antenna_pat_datasets(freq)]
            )
            if isinstance(product, nisarqa.SLC):
                calib_datasets = chain([calib_datasets, product.metadata_geometry_datasets()])
            if isinstance(product, nisarqa.RSLC):
                calib_datasets = chain([calib_datasets, product.metadata_crosstalk_datasets()])
            for ds in calib_datasets:
                has_finite &= _dataset_has_finite_pixels(ds)
                passes &= has_finite
                passes &= _dataset_is_not_all_zeros(ds)
                passes &= _check_gdal(product=product, ds=ds)
```